### PR TITLE
Github Action workflow base

### PR DESCRIPTION
**Pull request by giorgiozoppi**
Hello,
this is a skeleton CI workflow that allows to check if main is broken or not and ensure a minimal check coverage of 80 per cent. It's better of having nothing. As incremental we might add cppcheck and its custom rules for our use case both Misra/Autosar/cpp-core-guidelines to having better linted code,

### CODE REVIEW

**Review by stevenhartley — CHANGES_REQUESTED** [review]
@giorgiozoppi it appears that the majority of the commits you made are under the github email address that is not associated with the signed ECA. To fix this, go to your github account and then add your gm.com email address. The next day (not sure how long the cron job takes) recreate the PR with a single squashed commit and we can then take it from there. 

Thank you BTW for this!!!!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        sudo apt-get install -y libgtest-dev rapidjson-dev cgreen1  libspdlog-dev libcgreen1-dev wget lcov gawk
+    - name: Update Cmake to latest version.
+      run: |
+        sudo apt-get install ca-certificates gpg wget
+        wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | sudo tee /usr/share/keyrings/kitware-archive-keyring.gpg >/dev/null
+        echo 'deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ jammy main' | sudo tee /etc/apt/sources.list.d/kitware.list >/dev/null
+        sudo apt-get -y update && sudo apt-get -y install kitware-archive-keyring cmake
+ 
+    - name: Build Protocol Buffer.
+      run: |
```
Comment (stevenhartley) [inline .github/workflows/build.yml:49]: We should not be forced to build protobuf but use conan to pull build recipes.

**Comment by giorgiozoppi** [comment]
Ok. asap I've capacity I fix and help it out. Btw who is leading the rust effort?

**Comment by stevenhartley** [comment]
Daniel and Kai (from Etas & Bosch)

**Comment by giorgiozoppi** [comment]
close this. reopening during with a consistent account.

**Pull request closed without merging**